main.tf

```diff
@@ -425,7 +425,7 @@ resource "aws_cloudwatch_event_api_destination" "this" {
   invocation_endpoint              = each.value.invocation_endpoint
   http_method                      = each.value.http_method
   invocation_rate_limit_per_second = lookup(each.value, "invocation_rate_limit_per_second", null)
-  connection_arn                   = aws_cloudwatch_event_connection.this[each.value.name].arn
+  connection_arn                   = aws_cloudwatch_event_connection.this[try(each.value.connection_name, each.value.name)].arn
```

Comment:
Let's wrap with try() outside instead of inside the key. It will be easier to read.